tools/topology/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -30,47 +30,54 @@ DEBUG_START
 # PCM6 ---> volume <---- iDisp2
 # PCM7 ---> volume <---- iDisp3
 
-dnl PIPELINE_PCM_ADD(pipeline,
+dnl PIPELINE_PCM_DAI_ADD(pipeline,
 dnl     pipe id, pcm, max channels, format,
 dnl     period, priority, core,
-dnl     pcm_min_rate, pcm_max_rate, pipeline_rate,
-dnl     time_domain, sched_comp)
+dnl     dai type, dai_index, dai format,
+dnl     dai periods, pcm_min_rate, pcm_max_rate,
+dnl     pipeline_rate, time_domain)
 
 # Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-playback.m4,
 	1, 0, 2, s32le,
-	1000, 0, 0,
+	1000, 0, 0, ALH, 2, s32le, 3,
 	48000, 48000, 48000)
 
 # Low Latency capture pipeline 2 on PCM 1 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-capture.m4,
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-capture.m4,
 	2, 1, 2, s32le,
-	1000, 0, 0,
+	1000, 0, 0, ALH, 3, s32le, 3,
 	48000, 48000, 48000)
 
 # Low Latency playback pipeline 3 on PCM 2 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-playback.m4,
 	3, 2, 2, s32le,
-	1000, 0, 0,
+	1000, 0, 0, ALH, 0x102, s32le, 3,
 	48000, 48000, 48000)
 
 # Low Latency playback pipeline 4 on PCM 3 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-playback.m4,
 	4, 3, 2, s32le,
-	1000, 0, 0,
+	1000, 0, 0, ALH, 0x202, s32le, 3,
 	48000, 48000, 48000)
 
 # Low Latency capture pipeline 5 on PCM 4 using max 2 channels of s32le.
 # Schedule 48 frames per 1000us deadline on core 0 with priority 0
-PIPELINE_PCM_ADD(sof/pipe-volume-capture.m4,
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-capture.m4,
 	5, 4, 2, s32le,
-	1000, 0, 0,
+	1000, 0, 0, ALH, 0x302, s32le, 3,
```

Comment:
@tlauda is this change from PIPELINE_PCM_ADD to PIPELINE_PCM_DAI_ADD really necessary?

Reply:
@ranj063 Yes, because we need to change default number of periods to 3.

Reply:
I feel the direction is not quite right. We should have different definitions for pipelines and DAIs, and a connection at some point. Lumping everything in one giant macro is not exactly simple.

Reply:
@plbossart I know what you mean, but it would require major refactor of m4 scripts. This is definitely something we should think about. It took me quite some time to understand what, where and why.

Reply:
@tlauda I understand we want default number of periods set to 3. And that is already done below in the DAI_ADD() macro. Why do we need this in 2 different places? If you look at PIPELINE_PCM_DAI_ADD(), all it does is define the DAI_TYPE, index and periods. and setting these with the PIPELINE_PCM_ADD will only overwrite them without actually using it no?

Reply:
@ranj063 Periods are also used by every pipeline to set correct sink period count in components connected to DAI. The other possibility would be to define DAI first and pipeline right after that, but it would be even more complicated. We need to think about revamping the m4 scripts.